### ISSUE TITLE
no prefix for default locale

### ISSUE THREAD

**Issue by Aaike** [symfony-cmf/Routing#117]
I think there should be an option to specify the default locale and not have it prefixed.
so for the locales 'en' and 'de' if you specify 'en' as default you would get the following urls :
/home
/de/home

currenlty the locale is always prefixed for every language.
or maybe i missed that option somewhere ? i did have a good look through the documentation but couldn't find anything...

**Comment by hason** [comment]
@dbu What is status of this feature?

**Comment by dbu** [comment]
i am not aware of anything having been done. i think its mainly a documentation topic. the candidates strategy that looks for routes looks for both routes with and without the locale: https://github.com/symfony-cmf/Routing/blob/master/Candidates/Candidates.php#L89-L92

symfony will set the default locale if no locale comes out of the routing. so i would assume that nothing is missing. if it does not already work, please open a new issue and describe what you try to do and what does not work. (ah, if you try to make this happen with RoutingAuto i guess you would need something more - that would be an issue for https://github.com/symfony-cmf/RoutingAuto )

**Comment by nandorstanko** [comment]
@dbu Do you have some additional info on this one? I'm not sure is this a RoutingAuto issue or not, just found this open issue, probably it is related to my problem: https://github.com/symfony-cmf/routing-auto/issues/77

Basically I would want to achieve this:
somedomain.com - get homepage for default locale, without redirecting to somedomain.com/en
somedomain.com/about - get about page for default locale, without redirecting to somedomain.com/en/about
somedomain.com/de - get homepage for german locale

Maybe it's just a config issue, but I really don't know how to put all the pieces together...

**Comment by dbu** [comment]
hi @nandorstanko, https://github.com/symfony-cmf/routing-auto/issues/77 has nothing to do with what you are asking. it seems to be a limitation of the routing auto pattern. with the cmf router itself, / will find the root node of the routes tree. if that itself is a valid route, it works just fine.

for your question, i think all that is needed is storing the routes in the right location. the url is translated to a  repository path. if you have routes at /cms/routes, /cms/routes/about and /cms/routes/de your scenario should work.

are you using the routing auto or creating routes in an admin interface or with custom scripts? see also https://github.com/symfony-cmf/Routing/blob/master/Candidates/Candidates.php#L89-L92 for how potential routes are loaded.

**Comment by nandorstanko** [comment]
Hi @dbu! First, thank you for the response.
I'm creating the routes in an admin interface, basically the same way as in cmf-sandbox. I've figured out how to solve the routes for /about and /de/about, but still don't know how to get the route for the homepage content. In the cmf routing admin I can only add routes to some named nodes, like /en or /home. In the sandbox routing config there is a locale switcher which automatically redirects the / path to /en, but this is not the right way for me...
https://github.com/symfony-cmf/cmf-sandbox/blob/master/app/config/routing.yml#L10-L16

Probably there is some routing config combo where I can proxy the / to my /home route, but no success for me yet. I guess there is a simple solution for this, so if you have an idea, please share it with me :) Thanks!

**Comment by dbu** [comment]
`/` is the root node of the routes. if you configure nothing in routing.yml, you can use the document that is in phpcr. the routing bundle by default uses an initializer to place a generic node at the route root so that you can use that in sonata admin: https://github.com/symfony-cmf/routing-bundle/blob/master/src/Resources/config/initializer-phpcr.xml

i think the easiest way is to set `enable_initializer: false` in the routing bundle configuration and provide your own initializer to create a Route document that you store at the route root instead. i think that will also make the root route node show up in sonata.

**Comment by nandorstanko** [comment]
I guess I've figured this out now. The "problem" was, as you mentioned, that the root is a generic node, and not a Route node. With a custom initializer it is possible to change that. At the moment I've just added a home action to my default controller, which reads out the home content document, but I will try that more elegant solution also. Huge thanks for the help, and I think that a lot of people will learn from this conversation.
